src/util/Parameters.h

```diff
@@ -84,30 +86,40 @@ struct Parameter : public ParameterBase {
 
   /// Set the value.
   void set(Type newValue) {
-    _value = std::move(newValue);
+    if (parameterConstraint_.has_value()) {
+      std::invoke(parameterConstraint_.value(), newValue, name);
+    }
+    value_ = std::move(newValue);
     triggerOnUpdateAction();
   }
 
-  const Type& get() const { return _value; }
+  const Type& get() const { return value_; }
 
   /// Specify the onUpdateAction and directly trigger it.
   /// Note that this useful when the initial value of the parameter
   /// is known before the `onUpdateAction`.
   void setOnUpdateAction(OnUpdateAction onUpdateAction) {
-    _onUpdateAction = std::move(onUpdateAction);
+    onUpdateAction_ = std::move(onUpdateAction);
     triggerOnUpdateAction();
   }
 
+  /// Set an constraint that will be executed every time the value changes
+  /// and once initially when setting it.
```

Comment:
Also document that the way to use it is to throw an exception if the constraint is violated.